src/main/java/org/primeframework/mvc/util/QueryStringTools.java

```diff
@@ -40,7 +40,7 @@ public static Map<String, List<String>> parseQueryString(String query) {
     try {
       String[] parts = query.split("&");
       for (String part : parts) {
-        String[] pieces = part.split("=");
+        String[] pieces = part.split("=", 2);
```

Comment:
Did we see issues with this? I think that `=` in the value of a parameter should be encoded according to spec.

Reply:
It may be encoded, but it is not required. 

I don't know that it has ever bitten us yet, but it is not correct. This came up when reviewing how we are using the CSRF token in CleanSpeak. At times we take a value and append it to the URL as a query string after reading the value from the form. 

The value we inject into the form may have a suffix of `=` because of the base64 padding. From reviewing some of the changes in CleanSpeak I realized we aren't technically parsing this correctly. 

So in practice, for padding it won't matter because it is safe to remove, and in some cases it is being URL encoded to `%3d`.